R/dependencies.R

```diff
@@ -157,6 +157,28 @@ yamlDeps <- function(yaml) {
   )
 }
 
+stripAltEngines <- function(file, encoding) {
+  contents <- readLines(file, encoding = encoding)
+
+  # generate a list of all the headers
+  engineHeaders <- which(grepl("^## --.*engine=", contents))
+  allHeaders <- c(which(grepl("^## --", contents)), length(contents))
+
+  # calculate the end of each alternate engine code block (the beginning of the
+  # very next code block)
+  engineEnds <- vapply(engineHeaders, function(x) {
+    allHeaders[min(which(allHeaders > x))] - 1
+  }, 0)
+
+  # exclude the alternate engine code block lines
+  regions <- rep.int(TRUE, length(contents))
+  for (h in seq_along(engineHeaders)) {
+    regions[engineHeaders[[h]]:engineEnds[[h]]] <- FALSE
+  }
+
+  writeLines(contents[regions], file)
```

Comment:
Ack, of course, thanks -- I think I misread the code here thinking it was initialized based on the number of discovered engines rather than number of lines in the file.